### PR TITLE
Removes most of the enforced name rules

### PR DESCRIPTION
**Pull request by Recoherent**
## About The Pull Request
Guts most of the reject_bad_name proc to allow a larger degree of freedom when naming characters. This includes removal of automatic-capitalization and allows all species to use punctuation and numbers in their names. Sanitization of spaces still remains; this won't allow for massive blocks of spaces or leading/trailing whitespace or anything.

## Why It's Good For The Game
Ever got pissed that you couldn't put a number in your plasmaman's name, or that you couldn't name an agent card after your IPC?
This pretty obviously makes the name system more abusable, but that's the sort of thing that's easy to handle with admin intervention. I hold that the benefits of more open naming far outweigh opening it up to extremely obvious shitter behavior.

## Changelog
:cl:
tweak: Name enforcement is more open now; put those numbers on your agent card or your bizarre-super-soldier-clone-test-project-human's name!
code: Removed reference to allow_numbers from saved preferences.
/:cl:

### CODE REVIEW

**Comment by Calyxman** [comment]
'bizarre-super-soldier-clone-test-project-human's name!'
Wow you didn't have to be so direct

**Comment by Fikou** [comment]
> I hold that the benefits of more open naming far outweigh opening it up to extremely obvious shitter behavior.

I disagree, if you want to you can let agent cards use numbers, but I am not interested in this.

**Comment by Calyxman** [comment]
> > I hold that the benefits of more open naming far outweigh opening it up to extremely obvious shitter behavior.
> 
> I disagree, if you want to you can let agent cards use numbers, but I am not interested in this.

why can't the admins just manually enforce it?

**Comment by PrefabQuasar** [comment]
> > I hold that the benefits of more open naming far outweigh opening it up to extremely obvious shitter behavior.
> 
> I disagree, if you want to you can let agent cards use numbers, but I am not interested in this.

I see no reason to disallow human characters from using numbers in their names, which is essentially all this PR changes, you could already name yourself **incredibly** dumb things, and all things considered, its easier to police bad names than it is to have shitters prevented from outing themselves lurking on the server.

**Comment by tiramisuapimancer** [comment]
as possibly the lore head nobodys sure. this is based and cool actually

**Comment by Fikou** [comment]
> I see no reason to disallow human characters from using numbers in their names, which is essentially all this PR changes

that's not entirely right though, it lets everything no longer be capitalized and lets you put all base ascii in your name, this applies to a few more things than just character names as well.
i see no reason to remove functionality from the generic proc, when you can just use args for this.

i am also against unlimiting it in general but whatever, if the lore people want to let it for character names sure.

**Comment by ZephyrTFA** [comment]
I'm blocking this as a server issue due to the fact that names are almost never sanitized properly. The changes you have made to the name check proc allows html code through.
Until this is resolved I'm blocking this as a server breaking issue

**Comment by Recoherent** [comment]
Got it, didn't consider sanitization, thanks for bringing that up. I'll pull back to the previously allowed set of characters but keep the relaxed rules.

File: code/modules/client/preferences_savefile.dm [hunk 1]
```diff
@@ -461,14 +461,13 @@ SAVEFILE UPDATING/VERSIONING - 'Simplified', or rather, more coder-friendly ~Car
 		update_character(needs_update, S)		//needs_update == savefile_version if we need an update (positive integer)
 
 	//Sanitize
-	real_name = reject_bad_name(real_name, pref_species.allow_numbers_in_name)      // WS Edit - numbers in IPC names load from save
+	real_name = reject_bad_name(real_name)      // WS Edit - numbers in IPC names load from save
```
Comment (Fikou) [inline code/modules/client/preferences_savefile.dm]: remove the comment

**Review by Fikou — DISMISSED** [review]
This will make renaming robot parts limited to TRUE length (or 1), fix that

File: code/modules/client/preferences.dm [hunk 2]
```diff
@@ -2404,7 +2404,7 @@ GLOBAL_LIST_EMPTY(preferences_datums)
 		else
 			return
 	else
-		var/sanitized_name = reject_bad_name(raw_name,namedata["allow_numbers"])
+		var/sanitized_name = reject_bad_name(raw_name)
 		if(!sanitized_name)
 			to_chat(user, "<font color='red'>Invalid name. Your name should be at least 2 and at most [MAX_NAME_LEN] characters long. It may only contain the characters A-Z, a-z,[namedata["allow_numbers"] ? ",0-9," : ""] -, ' and .</font>")
```
Comment (Fikou) [inline code/modules/client/preferences.dm]: fix this.

**Review by ZephyrTFA — APPROVED** [review]
The issues I've indicated are resolved, however you still need to address Fikou's review

File: code/modules/client/preferences.dm (hunk 2 above)
Comment (Recoherent) [reply code/modules/client/preferences.dm]: for clarification, is this just for fixing the feedback message?

**Comment by Recoherent** [comment]
their review was addressed, just need them to sign off on it I suppose? just waiting until they're back

**Review by ZephyrTFA — APPROVED** [review]

**Pull request merged**